src/wast-parser.cc

```diff
@@ -1772,6 +1793,39 @@ Result WastParser::ParsePlainInstrVar(Location loc,
   return Result::Ok;
 }
 
+template <typename T>
+Result WastParser::ParseMemoryInstrVar(Location loc,
+                                       std::unique_ptr<Expr>* out_expr) {
+  Var memidx;
+	Var var;
+  if (PeekMatchLpar(TokenType::Memory)) {
+    if (!options_->features.multi_memory_enabled()) {
+      Error(loc, "Specifying memory variable is not allowed");
+      return Result::Error;
+    } else {
+      CHECK_RESULT(ParseMemidx(loc, &memidx));
+      CHECK_RESULT(ParseVar(&var));
+      out_expr->reset(new T(var, memidx, loc));
+      return Result::Ok;
```

Comment:
No need for this return since the it already exists at the outer level. 

Same for the blocks below.